go/vt/vtgate/planbuilder/testdata/union_cases.txt

```diff
@@ -1658,6 +1701,10 @@ Gen4 plan same as above
   "Original": "select col from unsharded union select id from user union select col2 from unsharded union select col from user_extra",
   "Instructions": {
     "OperatorType": "Distinct",
+    "Collations": [
+      "1: binary"
+    ],
+    "ResultColumns": 1,
     "Inputs": [
```

Comment:
The looks like the planner knows about collation, then the query should not contain weight_string column